src/main/java/com/battleships/battleships/ComputerPlayer.java

```diff
@@ -2,6 +2,7 @@
 
 import java.util.*;
 
+//
```

Comment:
Good that you've tried this advanced extension, and I can kind of see what you're trying to do.  However there's quite a lot of code here, and in the absence of test... ;-), I'm not going to get into it for now, I think we have some bigger structural issues to address first.